### PR TITLE
Delete workshops

### PR DESCRIPTION
**Pull request by Satttoshi**
pls check, added the D from cruD

### CODE REVIEW

File: backend/src/test/java/org/josh/backend/workshop/WorkshopControllerTest.java [hunk 1]
```diff
@@ -152,4 +152,45 @@ void expectUpdatedWorkshop_whenUpdateWorkshop() throws Exception {
...
+                MockMvcRequestBuilders.post("/api/workshop")
+                    .contentType(MediaType.APPLICATION_JSON)
+                    .content(testWorkshopWithoutIdAndLikes))
+            .andReturn().getResponse().getContentAsString();
+
+        Workshop saveResultWorkshop = objectMapper.readValue(result, Workshop.class);
+        String id = saveResultWorkshop.id();
+
+        //WHEN
+        mockMvc.perform(MockMvcRequestBuilders.delete("/api/workshop/%s".formatted(id)))
```
Comment (Kajochi) [inline backend/src/test/java/org/josh/backend/workshop/WorkshopControllerTest.java:171]: What does .formatted() do?

File: frontend/src/components/DeleteButton.tsx
Comment (Kajochi) [inline frontend/src/components/DeleteButton.tsx:1]: I like this!

File: frontend/src/hooks/useStore.ts [hunk 2]
```diff
@@ -48,7 +49,14 @@ export const useStore = create<State>((set, get) => ({
     updatePersonalStatus: (workshopId: string, personalStatus: PersonalStatus) => {
         axios.put(`/api/workshop/${workshopId}`, personalStatus)
             .catch(console.error)
-    }
+    },
+
+    deleteWorkshop: (workshopId: string) => {
+        const readWorkshops = get().readWorkshops;
```
Comment (Kajochi) [inline frontend/src/hooks/useStore.ts:55]: Why do you get the workshops here?

**Review by Kajochi — APPROVED** [review]

File: backend/src/test/java/org/josh/backend/workshop/WorkshopControllerTest.java (hunk 1 above)
Comment (Satttoshi) [reply backend/src/test/java/org/josh/backend/workshop/WorkshopControllerTest.java:171]: replaces the %s with a String. Use %i for integer and so on :v:

File: frontend/src/components/DeleteButton.tsx
Comment (Satttoshi) [reply frontend/src/components/DeleteButton.tsx:1]: thank you, if you want i can give you a quick Workshop with zustand!

File: frontend/src/hooks/useStore.ts (hunk 2 above)
Comment (Satttoshi) [reply frontend/src/hooks/useStore.ts:55]: i get a method from my "store" with get, so I can use it within this method.

The method does
1. get the read method
2. does the delete request
3. runs the read method (fetches the list again)

**Pull request merged**